crypto/src/keypair.rs

```diff
@@ -0,0 +1,18 @@
+use blsttc::{PublicKeyShare, SecretKeyShare};
+use hex;
+
+// KeyPair is a public key share and a secret key share that each node holds.
+pub struct KeyPair {
+    pub pk_share: PublicKeyShare,
+    pub sk_share: SecretKeyShare,
+}
+
+impl KeyPair {
+    pub fn hex_public_key(&self) -> String {
+        hex::encode(self.pk_share.to_bytes())
+    }
+
+    pub fn sign(&self, msg: &[u8]) -> String {
+        hex::encode(self.sk_share.sign(msg).to_bytes())
```

Comment:
Is there a way to verify partial signature? i.e., can we verify if a partial signature of node i is signed by node i?

Reply:
It is easily verified by `verify` function at `utils.rs`, provided the pubkey of node i. `test_verify` will help.